src/consumption.cpp

```diff
@@ -204,6 +204,21 @@ std::pair<int, int> Character::fun_for( const item &comest ) const
         }
     }
 
+    // Food is less enjoyable when eaten too often.
+    if( fun > 0 || comest.has_flag( "NEGATIVE_MONOTONY_OK" ) ) {
+        for( const consumption_event &event : consumption_history ) {
+            if( event.time > calendar::turn - 2_days && event.type_id == comest.typeId() &&
+                event.component_hash == comest.make_component_hash() ) {
```

Comment:
Since `comest.make_component_hash()` is always calculated here each time and not cached, you might just as well compare the component list directly, which actually might be faster, and you don't have to worry about hash collision and serialization too.

Reply:
I had it like that, but it would require storing the entire items in `event` which would be making a ton of unnecessary item copies.